gson/src/test/java/com/google/gson/GsonBuilderTest.java

```diff
@@ -254,4 +255,38 @@ public void testSetStrictness() throws IOException {
     assertThat(gson.newJsonReader(new StringReader("{}")).getStrictness()).isEqualTo(STRICTNESS);
     assertThat(gson.newJsonWriter(new StringWriter()).getStrictness()).isEqualTo(STRICTNESS);
   }
+
+  @Test
+  public void testRegisterTypeAdapterForObjectAndJsonElements() {
+    final String ERROR_MESSAGE = "Cannot override built-in adapter for ";
+    Type[] types = {
+            Object.class,
+            JsonElement.class,
+            JsonArray.class,
+    };
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    for (Type type : types) {
+      IllegalArgumentException e = assertThrows(IllegalArgumentException.class,
+              () -> gsonBuilder.registerTypeAdapter(type, NULL_TYPE_ADAPTER));
+      assertThat(e).hasMessageThat().isEqualTo(ERROR_MESSAGE + type);

```

Comment:
Really minor, but the indentation here still seems to be slightly incorrect:
```suggestion
    Type[] types = {
        Object.class,
        JsonElement.class,
        JsonArray.class,
    };
    GsonBuilder gsonBuilder = new GsonBuilder();
    for (Type type : types) {
      IllegalArgumentException e = assertThrows(IllegalArgumentException.class,
          () -> gsonBuilder.registerTypeAdapter(type, NULL_TYPE_ADAPTER));
      assertThat(e).hasMessageThat().isEqualTo(ERROR_MESSAGE + type);
```